### PR TITLE
Functional tests

### PR DESCRIPTION
**Pull request by ascott1**
This adds @marteki's functional test work with just a teeny bit of clean up and bug fixin' from me.
## To test

Pull this branch down and run `npm install`

Then make sure you have webdriver and protractir all set up:

``` bash
npm install -g webdriver-manager protractor
webdriver-manager update --standalone
```

To run the tests:

``` bash
# start webdriver in a terminal tab:
webdriver-manager start
# in a fresh tab:
protractor test/functional/conf.js
```
## Note

This requires a Node version 4+ or greater. We should either make note of this or use older versions of the testing modules to support older versions of Node. This isn't likely to be an issue anywhere other than a Jenkins server.
## Review
- @higs4281 
- @marteki 
- @niqjohnson

### CODE REVIEW

**Comment by niqjohnson** [comment]
:+1:  Woo!

```
Using the selenium server at http://localhost:4444/wd/hub
[launcher] Running 1 instances of WebDriver
Started
.....WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"costOfAttendance\"]") - the first result will be used
WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalCost\"]") - the first result will be used
.WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"costOfAttendance\"]") - the first result will be used
WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalCost\"]") - the first result will be used
.WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"costOfAttendance\"]") - the first result will be used
WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalCost\"]") - the first result will be used
.WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"costOfAttendance\"]") - the first result will be used
WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalCost\"]") - the first result will be used
.WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"costOfAttendance\"]") - the first result will be used
WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalCost\"]") - the first result will be used
.WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalGrantsScholarships\"]") - the first result will be used
WARNING - more than one element found for locator By.cssSelector("span[data-financial=\"totalCost\"]") - the first result will be used
.


11 specs, 0 failures
Finished in 24.482 seconds
[launcher] 0 instance(s) of WebDriver still running
[launcher] chrome #1 passed
```

**Comment by marteki** [comment]
Thank you for the help!!! And your changes explain why doing direct math in the tests wasn't working.

We might want to explicitly state that if folks are running this in standalone mode, they first need to run or make sure that `python manage.py runserver` is running in **a third and separate** tab before running the `protractor` command. (All the tabs are belong to us? Am I dating myself?)
**Update**: I added this to the branch.

That said, it's working for me, and I give a :+1: for merge.

File: test/functional/dd-functional-settlement-spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		expect( browser.getCurrentUrl() ).toContain( 'oid' );
+	} );
+
+	// TODO - Add expectation that other sections are invisible
+	it( 'should display the verify offer area first and no other sections', function() {
+		browser.wait( EC.visibilityOf(page.verifySection ), 8000 );
+	} );
+
+	// TODO - Add expectation that verification buttons disappear, and all fields that should be prepopulated actually are, and that Step 3 is still hidden
+	it("should let a student verify their information and go on to Step 1 and Step 2 of the offer", function() {
```
Comment (marteki) [inline test/functional/dd-functional-settlement-spec.js]: Found a missed quoting cleanup. I'll fix it and push to the branch.

**Comment by higs4281** [comment]
installs all work but i'm getting an error when running `protractor test/functional/conf.js`

``` bash
[launcher] Process exited with error code 1
undefined:1190
    vlog(2, () => this + ' scheduling notifications', this);
             ^
SyntaxError: Unexpected token )
```

**Comment by marteki** [comment]
Talked with @higs4281, and it's the same issue I ran into when setting up the tests initially: his Node version was at 0.12.*

@ascott1 Guess the Node 4.\* is an issue, even off a Jenkins server, and we should add an explicit note about that.

**Comment by higs4281** [comment]
:+1: after upgrading node (via brew in my case), tests run.

**Comment by marteki** [comment]
Added an explicit note about Node 4.0+ to the README, and merging. Onward to more tests! :dancer:

**Pull request merged**